jpeg/jpeg.go

```diff
@@ -0,0 +1,91 @@
+// jpeg handles scrubbing jpeg files of metadata
+package jpeg
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+
+	jpegimagestructure "github.com/dsoprea/go-jpeg-image-structure"
+)
+
+var headerBytes []byte = []byte{0xff, 0xd8}
+
+type JpegScrubber struct {
+	reader      io.Reader
+	segmentData io.LimitedReader
+	wroteHeader bool
+}
+
+func NewJpegScrubber(reader io.Reader) *JpegScrubber {
+	return &JpegScrubber{
+		reader: reader,
+	}
+}
+
+func isMetadataType(segmentMarker []byte) bool {
+	// TODO: consider other APP types
+	return segmentMarker[0] == 0xff &&
+		(segmentMarker[1] == jpegimagestructure.MARKER_APP1 || segmentMarker[1] == jpegimagestructure.MARKER_APP0)
+}
+
+func (self *JpegScrubber) Read(buf []byte) (n int, err error) {
```

Comment:
Receivers in Go are not typically named `self` or `this` ([here's a style guide of sorts which mentions this](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names)).  It's kind of Go dogmatism, but there are some justifications I guess.  [Here's a blog post about it](https://blog.heroku.com/neither-self-nor-this-receivers-in-go).  I disagree with the section on "Advanced Naming Technique" and actually the linter will flag that kind of thing nowadays.

A more idiomatic name for the receiver here would be `js`, `jpegScrubber`, or `scrubber`.

Reply:
arghhhh I am struggling to get used to this idiom, but thanks for pointing it out. I think you're right to point this out and I think in general it's probably better for me to get on board with this for the name of consistency.